sql/core/src/test/scala/org/apache/spark/sql/TypedImperativeAggregateSuite.scala

```diff
@@ -299,5 +319,87 @@ object TypedImperativeAggregateSuite {
     }
   }
 
+  /**
+   * Calculate the max value with object aggregation buffer. This stores class MaxValue
+   * in aggregation buffer.
+   */
+  private case class TypedMax2(
```

Comment:
can we simplify it? I think we just need to do some initialization work in  `createAggregationBuffer`.
```
case class MyUDAF ... {
  var initialized = false
  override def createAggregationBuffer(): MyBuffer = {
    initialized = true
    null
  }

  override def update(buffer: MaxValue, input: InternalRow): MyBuffer = {
    assert(initialized)
    null
  }

  ...
}
```